src/commands/register.ts

```diff
@@ -420,4 +424,38 @@ export default class ComponentRegister extends BaseCommand {
   public static getTagFromFlags(flags: any): string {
     return flags.environment ? `${ENV_TAG_PREFIX}${flags.environment}` : flags.tag;
   }
+
+  private async generateDependenciesWarnings(component_dependencies: Dictionary<string>, account_name: string) {
+    const dependency_arr: string[] = [];
+    for (const [component_name, tag] of Object.entries(component_dependencies)) {
+      dependency_arr.push(`${component_name}:${tag}`);
+    }
+    const dependencies: Dictionary<{ component: boolean, component_and_version: boolean }> = (await this.app.api.get(`accounts/${account_name}/components-tags`, { params: { components: dependency_arr } })).data;
+
+    const component_warnings: string[] = [];
+    const tag_warnings: string[] = [];
+    for (const [component_name, valid] of Object.entries(dependencies)) {
+      if (valid.component && valid.component_and_version) {
+        continue;
+      } else if (valid.component && !valid.component_and_version) {
+        tag_warnings.push(component_name);
+      } else {
+        component_warnings.push(component_name);
+      }
+    }
+
+    if (component_warnings.length > 0) {
+      this.log(chalk.yellow(`\nThe following dependencies need to be registered:`));
```

Comment:
Using `\n` should be avoided if possible. Would be better to just call `this.log()` with an empty string to make sure its platform safe.